### PR TITLE
Feature/update dependencies

### PR DESCRIPTION
**Pull request by gary-archer**
- Fixed Node.js vulnerabilities, use latest cloud native versions and fixed a couple of minor code and doc things.
- Also simplified use of SPIFFE helper now that you no longer need to build from source.

### CODE REVIEW

File: chapter-10-workload-identities/base/pki/spire/helm-values.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ spire-server:
   upstreamAuthority:
     certManager:
       enabled: true
```
Comment (gary-archer) [inline chapter-10-workload-identities/base/pki/spire/helm-values.yaml:22]: These are renamed in the latest SPIRE

File: chapter-10-workload-identities/base/pki/certmanager/rootca.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 ###################################################################################
 
 #
-# Create a self signed issuer for the root CA, in the cert-manager namespace
+# Create a self signed issuer for the internal root CA, in the cert-manager namespace
```
Comment (gary-archer) [inline chapter-10-workload-identities/base/pki/certmanager/rootca.yaml:7]: Make it clearer that these are internal certs

File: chapter-10-workload-identities/dbclient/README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ spec:
       - name: svids
         mountPath: /svids
   - name: dbclient-spiffehelper
-    image: dbclient-spiffehelper:v1
+    image: ghcr.io/spiffe/spiffe-helper:0.8.0
```
Comment (gary-archer) [inline chapter-10-workload-identities/dbclient/README.md:58]: There is now a Docker image so we don't need to build our own image which simplifies the code.

File: chapter-10-workload-identities/dbclient/dbclient.yaml [hunk 4]
```diff
@@ -23,17 +23,27 @@ spec:
...
       - name: dbclient
         image: dbclient:v1
+        securityContext:
+          {}
         volumeMounts:
           - name: svids
             mountPath: /svids
       - name: dbclient-spiffehelper
-        image: dbclient-spiffehelper:v1
+        image: ghcr.io/spiffe/spiffe-helper:0.8.0
```
Comment (gary-archer) [inline chapter-10-workload-identities/dbclient/dbclient.yaml:38]: Use the technique that [SPIRE uses](https://github.com/spiffe/helm-charts-hardened/blob/main/examples/mysql-using-spire/mysqlclient-statefulset.yaml#L50).

File: chapter-12-native-apps/desktop-app/package.json [hunk 5]
```diff
@@ -29,7 +29,7 @@
         "@types/node": "^20.11.30",
         "@types/react": "^18.3.2",
         "@types/react-dom": "^18.3.0",
-        "electron": "31.6.0",
+        "electron": "^33.2.0",
```
Comment (gary-archer) [inline chapter-12-native-apps/desktop-app/package.json:32]: Previously there was a macOS issue that caused tons of console output so I was using an old version. The issue is now fixed so we use latest again.

File: chapter-12-native-apps/desktop-app/src/main/ipcRequestHandler.ts [hunk 6]
```diff
@@ -119,7 +119,7 @@ export class IpcRequestHandler {
 
         try {
 
-            if (!event.senderFrame.url.startsWith('file:/')) {
+            if (!event.senderFrame?.url.startsWith('file:/')) {
```
Comment (gary-archer) [inline chapter-12-native-apps/desktop-app/src/main/ipcRequestHandler.ts:122]: There was a possibly null code warning

**Review by mtrojanowski — APPROVED** [review]

**Pull request merged**